### PR TITLE
fix build and add LLVM and Clang to Prerequisites

### LINKED ISSUE #346: Build error when generate bindings to gRPC in CentOS
**Describe the bug**
Build error when generate bindings to gRPC in CentOS 7
```
HOST = Some("x86_64-unknown-linux-gnu")
CXX_x86_64-unknown-linux-gnu = None
CXX_x86_64_unknown_linux_gnu = None
HOST_CXX = None
CXX = None
CXXFLAGS_x86_64-unknown-linux-gnu = None
CXXFLAGS_x86_64_unknown_linux_gnu = None
HOST_CXXFLAGS = None
CXXFLAGS = None
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("true")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
running: "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "grpc/include" "-Wall" "-Wextra" "-std=c++11" "-DGRPC_SYS_SECURE" "-Werror" "-o" "/home/lxt/Projects/grpc-rs/target/debug/build/grpcio-sys-3e86be41c3f6dd46/out/grpc_wrap.o" "-c" "grpc_wrap.cc"
exit code: 0
AR_x86_64-unknown-linux-gnu = None
AR_x86_64_unknown_linux_gnu = None
HOST_AR = None
AR = None
running: "ar" "crs" "/home/lxt/Projects/grpc-rs/target/debug/build/grpcio-sys-3e86be41c3f6dd46/out/libgrpc_wrap.a" "/home/lxt/Projects/grpc-rs/target/debug/build/grpcio-sys-3e86be41c3f6dd46/out/grpc_wrap.o"
exit code: 0
cargo:rustc-link-lib=static=grpc_wrap
cargo:rustc-link-search=native=/home/lxt/Projects/grpc-rs/target/debug/build/grpcio-sys-3e86be41c3f6dd46/out
CXXSTDLIB_x86_64-unknown-linux-gnu = None
CXXSTDLIB_x86_64_unknown_linux_gnu = None
HOST_CXXSTDLIB = None
CXXSTDLIB = None
cargo:rustc-link-lib=stdc++

--- stderr
./grpc/include/grpc/impl/codegen/slice.h:24:10: fatal error: 'stddef.h' file not found
./grpc/include/grpc/impl/codegen/slice.h:24:10: fatal error: 'stddef.h' file not found, err: true
thread 'main' panicked at 'Unable to generate grpc bindings: ()', src/libcore/result.rs:999:5
```

**Method that has been tried**
1. scl enable llvm-toolset-7
2. yum update cmake 
3. yum update gcc/g++
4. use a new OS to build in strict accordance with the documentation

### PR DESCRIPTION
**Pull request by light4**
Fixes #346

### CODE REVIEW

File: grpc-sys/Cargo.toml [hunk 1]
```diff
@@ -59,7 +59,7 @@ no-omit-frame-pointer = []
 
 [build-dependencies]
 cc = "1.0"
-cmake = "0.1.27"
+cmake = "0.1.40"
 pkg-config = "0.3"
-bindgen = "0.49"
-walkdir = "2.2.8"
\ No newline at end of file
+bindgen = "0.50.0"
```
Comment (BusyJay) [inline grpc-sys/Cargo.toml]: Can you customize the features so that clap is opted out?

**Comment by BusyJay** [comment]
I can verify that it can be compiled without any custom environment variables. @hunterlxt PTAL

File: README.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ This project is still under development. The following features with the check m
 - CMake >= 3.8.0
 - Rust >= 1.19.0
 - binutils >= 2.22
+- LLVM and Clang used by bindgen
```
Comment (hunterlxt) [inline README.md]: Can you clearly state the minimum version we need?

**Comment by hunterlxt** [comment]
> I can verify that it can be compiled without any custom environment variables. @hunterlxt PTAL

Just install packages required via `yum` ? I try it again and loos like it didn't work.

**Comment by BusyJay** [comment]
> Just install packages required via yum ?

Yes. I tried it on fedora 30.

File: README.md (hunk 2 above)
Comment (light4) [reply README.md]: I'm not sure, but LLVM and Clang with version 3.4.2 is fine. See https://github.com/pingcap/grpc-rs/issues/346#issuecomment-513187882.

**Comment by aknuds1** [comment]
This fixes #344 too AFAICT! Thanks!

File: README.md (hunk 2 above)
Comment (BusyJay) [reply README.md]: Can you add the version number to readme?

Comment (light4) [reply README.md]: Just use the version from CentOS7 repo.

**Comment by Hoverbear** [comment]
It looks like the CI failure is related to unrelated Windows Chocolatey problems. I think we can merge this without seeing that particular build pass for now.

cc @nrc @BusyJay

File: README.md (hunk 2 above)
Comment (BusyJay) [reply README.md]: I just checked the docs from rust-bindgen, seems >3.9 is recommended. https://rust-lang.github.io/rust-bindgen/requirements.html

Comment (light4) [reply README.md]: updated

**Comment by BusyJay** [comment]
CI should be fixed by #351.

**Comment by BusyJay** [comment]
@light4 do you have time to fix the conflict?

**Comment by BusyJay** [comment]
I solved the conflict for you. Once all CIs except windows pass, we can merge it.

**Pull request merged**